File: 230.Kth.Smallest.Element.in.a.BST.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int kthSmallest(TreeNode* root, int k) {
        auto node = root;
        stack<TreeNode*>st;
        while(1)
        {
            if(node)
            {
                st.push(node);
                node = node->left;
            }
            else if (!st.empty())
            {
                node = st.top(); st.pop();
                k--;
                if(k==0)
                {
                    return node->val;
                }
                node = node->right;
            }
        }
        return 0;
    }
};

File: 230. Kth Smallest Element in a BST.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: TreeNode, k: int) -> int:
        q = []
        
        def append_nodes(node: TreeNode):
            while node:
                q.append(node)
                node = node.left
        
        append_nodes(root)
        while len(q) > 0:
            node = q.pop()
            if (k := k-1) == 0:
                return node.val
            append_nodes(node.right)
    